File: src/Services/Roslyn/Markify.Services.Roslyn.Fixtures/Projects/Source/Types/Partial.cs
using System;
using System.Collections;

#region Class

namespace Class
{
    public sealed partial class FooType : IDisposable
    {
        private int _fieldOne;

        public event EventHandler Done;

        private int PropertyOne { get; set; }

        partial void PartialMethod();

        public void MethodOne(int foo)
        {
        }
    }

    public partial class FooType : IEnumerable
    {
        private float _fieldTwo;

        private int PropertyTwo { get; set; }

        public event EventHandler Started;

        partial void PartialMethod()
        {
        }

        partial void PartialMethod(int foo);

        public void MethodTwo(int foo, int bar)
        {
        }
    }
}

#endregion

#region Interface

namespace Interface
{
    public partial interface FooType : IDisposable
    {
        int PropertyOne { get; set; }

        event EventHandler Done;

        void MethodFour(string foo);

        void MethodOne(int foo);
    }

    public partial interface FooType : IEnumerable
    {
        int PropertyTwo { get; set; }

        event EventHandler Started;

        void MethodTwo(int foo, int bar);

        void MethodThree(double foo);
    }
}

#endregion

#region Struct

namespace Struct
{
    public partial struct FooType : IDisposable
    {
        private int _fieldOne;

        private int PropertyOne { get; set; }

        public event EventHandler Done;

        partial void PartialMethod();

        public void MethodOne(int foo);
    }

    public partial struct FooType : IEnumerable
    {
        private float _fieldTwo;

        private int PropertyTwo { get; set; }

        public event EventHandler Started;

        partial void PartialMethod() { }

        partial void PartialMethod(int foo);

        public void MethodTwo(int foo, int bar) { }
    }
}

#endregion

File: src/Services/Roslyn/Markify.Services.Roslyn.Fixtures/Projects/Source/Types/Partial.vb
Imports System
Imports System.Collections

#Region "Class"

Namespace Class
    Partial Public NotInheritable Class FooType
        Implements IDisposable
        Private _fieldOne As Integer

        Private Property PropertyOne As Integer

        Public Event Done As EventHandler

        Partial Sub PartialMethod()
        End Sub

        Public Sub MethodOne(foo As Integer)
        End Sub
    End Class

    Partial Public Class FooType
        Implements IEnumerable
        Private _fieldTwo As Decimal

        Private Property PropertyTwo As Integer

        Public Event Started As EventHandler

        Partial Sub PartialMethod()
        End Sub

        Partial Sub PartialMethod(foo As Integer)
        End Sub

        Public Sub MethodTwo(foo As Integer, bar As Integer)
        End Sub
    End Class
End Namespace

#End Region

#Region "Interface"

Namespace Interface
    Partial Public Interface FooType
        Inherits IDisposable
        Property PropertyOne As Integer

        Event Done As EventHandler

        Sub MethodOne(foo As Integer)

        Sub MethodFour(foo As String)
    End Interface

    Partial Public Interface FooType
        Inherits IEnumerable
        Property PropertyTwo As Integer

        Event Started As EventHandler

        Sub MethodTwo(foo As Integer, bar As Integer)

        Sub MethodThree(foo As Double)
    End Interface
End Namespace

#End Region

#Region "Struct"

Namespace Struct
    Partial Public Structure FooType
        Implements IDisposable
        Private _fieldOne As Integer

        Private Property PropertyOne As Integer

        Public Event Done As EventHandler

        Partial Sub PartialMethod()
        End Sub

        Public Sub MethodOne(foo As Integer)
        End Sub
    End Structure

    Partial Public Structure FooType
        Implements IEnumerable
        Private _fieldTwo As Decimal

        Private Property PropertyTwo As Integer

        Public Event Started As EventHandler

        Partial Sub PartialMethod()
        End Sub

        Partial Sub PartialMethod(foo As Integer)
        End Sub

        Public Sub MethodTwo(foo As Integer, bar As Integer)
        End Sub
    End Structure
End Namespace

#End Region